lib/trie/trie.go

```diff
@@ -324,9 +329,9 @@ func (t *Trie) insert(parent Node, key []byte, value Node) Node {
 		}
 
 		return br
+	default:
+		panic("unknown node type: " + fmt.Sprint(newParent.Type()))
```

Comment:
Why do we want to panic rather than return an error here?

Reply:
If this happens we have something terribly wrong coded :smile: We should really panic instead of trying to recover and continue things. Maybe that can be removed with generics in Go 1.18 or (as Kishan suggested) if we merge both leaf and branch in the same structure and don't use interfaces. But yeah not for now :wink:

Reply:
yeah this case should never be hit, we just have to have it to make the compiler happy afaik